chromadb/utils/embedding_functions/openai_embedding_function.py

```diff
@@ -92,6 +86,21 @@ def __init__(
         self._deployment_id = deployment_id
 
     def __call__(self, input: Documents) -> Embeddings:
+        """
+        Generate the embeddings for the given `input`.
+
+        # About ignoring types
```

Comment:
I realise now that one of the implementations does `e.index` and the other does `e["index"]`. I wasn't sure that the latter was allowed, so maybe we can cast the types as done [here](https://github.com/chroma-core/chroma/pull/2034/commits/6ad759829df125d17e6b26d20ebb987b70861705#r1573627153),

Reply:
Does it make sense to use dynamic imports from the lib?

- v1 - https://github.com/openai/openai-python/blob/main/src/openai/types/create_embedding_response.py#L20
- v0 we can maybe cast to `Dict[str,Any]`

Reply:
I'm not quite sure what you mean by the first, can you be more explicit? Is there anywhere in the codebase where I can take a look? 
As for the `Dict[str,Any]` it throws me this:

```
Argument 1 to "sorted" has incompatible type "Dict[str, Any]"; expected "Iterable[Dict[str, Union[str, List[float]]]]"
```
Which points to what I did in [jina](https://github.com/chroma-core/chroma/pull/2034/commits/6ad759829df125d17e6b26d20ebb987b70861705#r1573627153)

Also, removing the type hint ignore, will raise additional annoying issues as it says that `Name "embeddings" already defined on line 124  [no-redef]` so we will be forced to tweak actual code we don't have tests for. For me it feels a call for potential issues :thinking: